src/main/java/com/google/cloud/flink/bigquery/BigQueryDirectDataWriterContext.java

```diff
@@ -0,0 +1,219 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.flink.bigquery;
+
+import static com.google.cloud.flink.bigquery.ProtobufUtils.buildSingleRowMessage;
+import static com.google.cloud.flink.bigquery.ProtobufUtils.toDescriptor;
+
+import com.google.api.gax.retrying.RetrySettings;
+import com.google.api.gax.retrying.RetrySettings.Builder;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryException;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.Field.Mode;
+import com.google.cloud.bigquery.FieldList;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.StandardSQLTypeName;
+import com.google.cloud.bigquery.StandardTableDefinition;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.bigquery.TableInfo;
+import com.google.cloud.bigquery.TimePartitioning;
+import com.google.cloud.bigquery.connector.common.BigQueryClientFactory;
+import com.google.cloud.bigquery.connector.common.BigQueryConfig;
+import com.google.cloud.bigquery.connector.common.BigQueryConnectorException;
+import com.google.cloud.bigquery.connector.common.BigQueryCredentialsSupplier;
+import com.google.cloud.bigquery.storage.v1beta2.ProtoSchema;
+import com.google.cloud.flink.bigquery.common.BigQueryDirectDataWriterHelper;
+import com.google.cloud.flink.bigquery.common.BigQueryDirectWriterCommitMessageContext;
+import com.google.cloud.flink.bigquery.common.DataWriterContext;
+import com.google.cloud.flink.bigquery.common.UserAgentHeaderProvider;
+import com.google.cloud.flink.bigquery.common.WriterCommitMessageContext;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Descriptors;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import net.sf.jsqlparser.JSQLParserException;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.table.types.logical.StructuredType;
+import org.apache.flink.table.types.logical.StructuredType.StructuredAttribute;
+import org.apache.flink.types.Row;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BigQueryDirectDataWriterContext implements DataWriterContext<Row> {
+
+  final Logger logger = LoggerFactory.getLogger(BigQueryDirectDataWriterContext.class);
+
+  final String tablePath;
+  final StructuredType flinkSchema;
+  final Descriptors.Descriptor schemaDescriptor;
+  static BigQuery bigquery;
+  private BigQueryDirectDataWriterHelper writerHelper;
```

Comment:
Please name this writeHelper and not writerHelper

Reply:
Taken care of naming correction